In [3]:
# Your task in this week’s assignment is to answer three questions:
# 1. What is the northernmost airport in the United States?
# 2. What is the easternmost airport in the United States?
# 3. On February 12th, 2013, which New York area airport had the windiest weather?

# Being successful on this assignment requires using both:
# • Your developing skills in Python and pandas, and
# • Your research skills in validating your candidate answers in context.

# Tasks
# 1. Use the information in the .CSV files found at
# https://github.com/hadley/nycflights13/tree/master/data-raw for your source data.
# 2. Write your code using Python and pandas, and deliver it in a Jupyter notebook. You should post
# your notebook in a GitHub repository, and send me a link to the notebook.
# 3. For each of the three questions listed above, you’ll need to verify your answers (in two cases by
# searching for additional confirming information, and in the third case being alert for possible
# outliers). For the first two questions above, you may want to start by generating a list of the five
# northernmost or easternmost U.S. airports.
# 4. In addition to providing code, you should state your answers to each of the above questions (in
# Markdown) in the Jupyter notebook.
# 5. To receive full credit, you’ll need to include in your Jupyter notebook both your code and your
# documented assumptions and any changes that you made to the data based on your research
# (using Markdown) in your Jupyter notebook.

# Saar Turjeman
# IS362 - assignment# 5
# SPRING2024

%pylab inline
import pandas as pd
from pandas import Series, DataFrame

In [42]:
# import the airlines DataFrame, downloaded from GitHub
airlines = pd.read_csv('airlines.csv')

# import the airports DataFrame, downloaded from GitHub
airports = pd.read_csv('airports.csv')

# import the planes DataFrame, downloaded from GitHub
planes = pd.read_csv('planes.csv')

# import the weather DataFrame, downloaded from GitHub
weather = pd.read_csv('weather.csv')

# case 1

# Latitude is the primary determinant of northness or southness. The higher the latitude, the more north the location is

In [48]:
# 1. What is the northernmost airport in the United States?


# displays the airports DataFrame, and sorting the values in lat in descending order
# and shows the top 5 values using the .head() method
airports.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

# case 2

# Checking based on Longitude 

# Longitude (lon): Longitude primarily indicates the east-west position of a location and does not directly relate to its northness. However, if latitude were the same between two locations, the one with the less negative longitude (or more positive) would be further east and therefore considered more north in relation to the same latitude.

In [50]:
airports.sort_values('lon', ascending=False).head()

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York

# case 3 - outlier
# getting fact from the web

### according to Google search BRW is the norternmost airport in the us

https://ourairports.com/countries/US/AK/


### according to online sources EEN lat is not 72.270833, it's lat is 42.53

https://latitude.to/articles-by-country/us/united-states/179120/dillanthopkins-airport

https://keenenh.gov/sites/default/files/2017-11/Final%20Environmental%20Assessment%20Dillant-Hopkins%20Airport.pdf

# answer to question# 1

In [98]:
# the answer for question # 1 - BRW
# 1. What is the northernmost airport in the United States?
sorted_airports = airports.sort_values('lat', ascending=False)
northernmost_airport = sorted_airports.iloc[1:2]
northernmost_airport

faa                        name        lat         lon  alt  tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44  -9   A   

                 tzone  
230  America/Anchorage

In [80]:
# 2. What is the easternmost airport in the United States?

# Easternmost Airport:
# The higher the longitude the more east the airport is located.

# Sorting the values in the lon table showing the first five airports with the highest longitude


# case 1
### For the lon 174.113620: Despite being a positive longitude, it represents a location more westward, likely in the western hemisphere.
### For -67.0129694: Being a negative longitude, it signifies a location west of the Prime Meridian, making it more eastward compared to the positive longitude value provided. Therefore, -67.0129694 (likely in Maine) is more easternmost compared to 174.113620.


In [81]:
airports.sort_values('lon', ascending=False).head(5)

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York

# case 2
### I wanted to look by tz or tzone, but could not find all the cities listed. 
### missing cities from eastern states!

In [83]:
airports.sort_values('tz', ascending=False).head(5)

faa                           name        lat         lon   alt  tz dst  \
942  MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396  DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
0    04G              Lansdowne Airport  41.130472  -80.619583  1044  -5   A   
779  LDJ                 Linden Airport  40.617447  -74.244594    23  -5   A   
707  JZP         Pickens County Airport  34.453472  -84.457222  1535  -5   A   

                tzone  
942    Asia/Chongqing  
396    Asia/Chongqing  
0    America/New_York  
779  America/New_York  
707  America/New_York

# case 3
### confirmed longitude on the internet

https://www.airport-data.com/airport/EPM/

# answer to question# 2

In [102]:
# the answer for question # 2 - EPM
# 1. What is the Easternmost airport in the United States?
sorted_airports = airports.sort_values('lon', ascending=False)
easternmost_airport = sorted_airports.iloc[4:5]
easternmost_airport

faa                        name        lat        lon  alt  tz dst  \
444  EPM  Eastport Municipal Airport  44.910111 -67.012694   45  -5   A   

                tzone  
444  America/New_York

In [103]:
# 3. On February 12th, 2013, which New York area airport had the windiest weather?

In [110]:
weather.sort_values('wind_speed',ascending=False).head(5)

origin  year  month  day  hour   temp   dewp   humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96   61.63     260.0   
9425     JFK  2013      1   31     4  53.60  53.06  100.00     200.0   
724      EWR  2013      1   31     6  57.20  53.60   87.74     270.0   
722      EWR  2013      1   31     4  60.80  59.00   93.79     230.0   
18131    LGA  2013      1   31     4  59.00  55.40   93.69     230.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN    0.00    1008.3   10.0  2013-02-12T08:00:00Z  
9425     42.57886   52.93588    0.20       NaN    2.0  2013-01-31T09:00:00Z  
724      42.57886   55.23744    0.03       NaN   10.0  2013-01-31T11:00:00Z  
722      40.27730   58.68978    0.30       NaN    6.0  2013-01-31T09:00:00Z  
18131    40.27730   55.23744    0.41       NaN    8.0  2013-01-31T09:00:00Z

In [113]:
feb12_2013 = weather[(weather['month'] == 2) & (weather['day'] == 12) & (weather['year'] == 2013)]
feb12_2013.sort_values('wind_speed', ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z

# answer to question# 3

In [115]:
# highest wind speed 
feb12_2013.sort_values('wind_speed', ascending=False).head(1)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009    EWR  2013      2   12     3  39.02  26.96  61.63     260.0   

      wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009  1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z